src/Dropdown.tsx

```diff
@@ -1,47 +1,61 @@
 /** @jsxImportSource @emotion/react */
 import { css } from "@emotion/react";
 import { useEffect } from "@storybook/addons";
-import { createContext, ReactNode, useContext, useState } from "react";
-
-interface SelectOption {
-  // children: ReactNode;
-  // value?: string | null;
-  label: string;
-}
+import {
+  createContext,
+  MouseEvent,
+  ReactNode,
+  useContext,
+  useState,
+} from "react";
+
+type SelectStatusContext = {
+  open?: Boolean;
+  setOpen?: (data: Boolean) => void;
+  optionUI?: Boolean;
+  setOptionUI: (data: Boolean) => void;
+  selected: String;
+  setSelected?: (data: String) => void;
+  selectPlaceholder?: String;
+  updateSelected: (data: String) => void;
+  handleButton: () => void;
+};
 
-interface ISelectContext {
-  open: Boolean;
-  setOpen: (data: Boolean) => void;
+interface SelectProps {
+  children: ReactNode;
+  onChange: (data: any) => void;
+  defaultValue: String;
+  placeholder: String;
+  values: String[];
 }
 
-const SelectContext = createContext<ISelectContext | undefined>();
+const SelectContext = createContext<SelectStatusContext | null>(null);
 
 // typescript는 조금 더 공부 하고 추가하겠습니다.
-const Select: React.FC = ({
+const Select = ({
   children,
   onChange,
   defaultValue,
   placeholder,
-}) => {
-  const [open = false, setOpen] = useState(false);
-  const [selected, setSelected] = useState(defaultValue);
+}: SelectProps) => {
+  const [open = false, setOpen] = useState<Boolean>(false);
```

Comment:
```suggestion
  const [open = false, setOpen] = useState(false);
```

`open`이라는 친구는 boolean 이라는 타입으로 추론(infer)이 된다.